repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -3481,6 +3490,76 @@ public void deleteClassificationOnlyPropagation(Set<String> deletedEdgeIds) thro
         }
     }
 
+    public void deleteClassificationOnlyPropagation(String deletedEdgeId) throws AtlasBaseException {
```

Comment:
Why we are executing this again with only one edge, classification Id is missing here. Again we are fetching all the classification vertices and same thing quite exponentially.
Like Edge 1 has A,B attached we create task (1,A)(1,B) inside each task again we are fetching 1,2 and looping over them, that is a redundant operation. Just add a check that if the current classification attached has the classification id attached.

Reply:
You are correct. Addressed now. Missed this while cherry picking

Reply:
In terms of adding a check if the classification already attached to edge or not, don't we need to check for all related vertices if that classification is attached to downstream assets? If I were to check if classification is attached or not and them not executing the operation at all when classification is not attached, I have a concern that maybe downstream assets need to be checked as well. what do you think?

Reply:
So, basically if we want to check for all downstream it would same as traversing all the downstream, just check that if classification is attached to the end of edge mean all the downstream has classification deleted and vice versa. e.g A-> B edge, here if classification is attached to B mean its downstream has the classification, if not then we can consider by other task all the downstream classifications were removed. You can just use `GraphHelper.getPropagatableClassifications` to check that, if classification id is not present in propagable classification then we can skip this. Verify from your end also.

Reply:
Can we just assume that if classification is not attached to the nearest downstream entity then we can say that it is not attached to all downstream entities?

Reply:
This is the current implementation actually we get the propagable classification by that logic and start executing if  classification is attached to the nearest downstream entity

Reply:
> start executing if classification is attached to the nearest downstream entity
I do not think we do have this check in only delete task execution in the moment and also you are suggesting to put one. 

But my concern is whether we should look for the nearest downstream entity if all of the downstream entities have the same situation or not